src/py/f_table.rs

```diff
@@ -2,14 +2,15 @@ use std::collections::HashMap;
 
 use crate::allocator::Allocator;
 use crate::core_ops::{op_cons, op_eq, op_first, op_if, op_listp, op_raise, op_rest};
+use crate::cost::Cost;
 use crate::more_ops::{
     op_add, op_all, op_any, op_ash, op_concat, op_div, op_divmod, op_gr, op_gr_bytes, op_logand,
     op_logior, op_lognot, op_logxor, op_lsh, op_multiply, op_not, op_point_add, op_pubkey_for_exp,
     op_sha256, op_softfork, op_strlen, op_substr, op_subtract,
 };
 use crate::reduction::Response;
 
-type OpFn<T> = fn(&mut T, <T as Allocator>::Ptr) -> Response<<T as Allocator>::Ptr>;
+type OpFn<T> = fn(&mut T, <T as Allocator>::Ptr, Cost) -> Response<<T as Allocator>::Ptr>;
```

Comment:
If you make this an `enum` with two flavors of function signature, one of which doesn't take a second parameter, you can avoid making all the gratuitous signature changes to the functions that can't possibly overflow.

Reply:
Might not be worth it, not sure. But it's definitely a possibility.

Reply:
yeah, I suspect it would end up being more complex overall though. an unused parameter is pretty simple and straight-forward (and no need to store a discriminator next to each function pointer)